### PR TITLE
Fix for CORE-2071 via changes to `pgp pull`

### PR DESCRIPTION
**Pull request by maxtaco**
Now `pgp pull` works when logged out, and runs a track before importing the keys in GPG

@oconnor663 / @patrickxb  what do you think?

@strib this should solve your issue

### CODE REVIEW

File: go/engine/pgp_pull.go [hunk 1]
```diff
@@ -128,20 +127,71 @@ func (e *PGPPullEngine) getTrackedUserSummaries(ctx *Context) ([]keybase1.UserSu
 	return matchedList, nil
 }
 
-func (e *PGPPullEngine) Run(ctx *Context) error {
-	summaries, err := e.getTrackedUserSummaries(ctx)
+func (e *PGPPullEngine) runLoggedOut(ctx *Context) error {
+	if len(e.userAsserts) == 0 {
```
Comment (oconnor663) [inline go/engine/pgp_pull.go]: Does this not also get checked in the logged in version? What does it mean to pull a PGP key with no user?

Comment (maxtaco) [inline go/engine/pgp_pull.go]: Pull for all trackers.

Comment (oconnor663) [inline go/engine/pgp_pull.go]: Ok now I remember. It's "get everyone's updated PGP keys"?

Comment (maxtaco) [inline go/engine/pgp_pull.go]: Yes

File: go/engine/pgp_pull.go [hunk 2]
```diff
@@ -160,19 +210,26 @@ func (e *PGPPullEngine) Run(ctx *Context) error {
...
-			err = gpgClient.ExportKey(*bundle)
-			if err != nil {
-				return err
-			}
-			ctx.LogUI.Info("Imported key for %s.", user.GetName())
+func (e *PGPPullEngine) exportKeysToGPG(ctx *Context, user *libkb.User, tfp map[string]bool) error {
+	for _, bundle := range user.GetActivePGPKeys(false) {
+		// Check each key against the tracked set.
+		if tfp != nil && !tfp[bundle.GetFingerprint().String()] {
+			ctx.LogUI.Warning("Keybase says that %s owns key %s, but you have not tracked this fingerprint before.", user.GetName(), bundle.GetFingerprint())
```
Comment (oconnor663) [inline go/engine/pgp_pull.go]: Old code, but I'm wondering if this warning makes sense. We really should be doing the right thing with each user's sigchain, and checking the proofs that they own these keys. If that means the "fetch everyone" version takes forever, so be it. How often could you possibly need to run that command, that you can't let it sit in the background for a few minutes?

**Comment by oconnor663** [comment]
This looks good to me, other than my thought at the end there which isn't to do with your changes.

**Pull request merged**